### ISSUE TITLE
Avoid Duplicate Reporting of Lockfile Poisoning in Markdown Summary Reporter

### ISSUE THREAD

**Issue by abhisek** [safedep/vet#231]
This issue is manifested when there are a lot of packages from private repositories. Although we have a config option to add private repository URL to allow list, it makes sense to consolidate the finding by `URL`

Reference: https://github.com/safedep/vet-action/issues/37#issuecomment-2275005354

**Comment by AmalChandru** [comment]
@abhisek Is this still up for grabs?

**Comment by abhisek** [comment]
@AmalChandru Yes. This issue still exists. While it was triggered in `vet-action`, the root cause is in `vet`.

Specifically in `--report-markdown-summary`

https://github.com/safedep/vet/blob/main/pkg/reporter/markdown_summary.go
